platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -543,19 +525,31 @@ void flushUnlock() {
      * critical for snapshots (e.g. update an index), it will be stopped until {@link
      * #resumeMerging()}} is called.
      */
-    void pauseMerging() {
-        mergingPaused.acquireUninterruptibly();
+    void pauseMerging() throws IOException {
+        if (hasDiskStoreForInternalHashes) {
+            internalHashStoreDisk.pauseMerging();
+        }
+        pathToHashKeyValue.pauseMerging();
+        if (!isLongKeyMode) {
+            objectKeyToPath.pauseMerging();
+        }
     }
 
     /** Resumes previously stopped data file collection merging. */
-    void resumeMerging() {
-        mergingPaused.release();
+    void resumeMerging() throws IOException {
+        if (hasDiskStoreForInternalHashes) {
+            internalHashStoreDisk.resumeMerging();
+        }
+        pathToHashKeyValue.resumeMerging();
+        if (!isLongKeyMode) {
+            objectKeyToPath.resumeMerging();
+        }
     }
 
     /**
      * Save a batch of data to data store.
      *
-     * <p>If you call this method where not all data is provided to cover the change in
+     * If you call this method where not all data is provided to cover the change in
```

Comment:
Quick question on these reformat items.  Was this from the IDE or `./gradlew spotlessApply`?  It is odd that spotless would reformat this much on a codebase that was already reformatted, and if it continues to do so we may need to make some changes in the configuration.

Reply:
One source of frustration was that the PR (not this one, but the one in the swirdls-platform repo) was very old and had outdated spotless config when I ran it for the first time. Some of these <p> tags were added then. After that, I synced with develop, got updated spotless rules into the branch, and had to re-run spotless again, which also contributed to the mess... The end result is there may be some tags added or removed for no good reason.

Reply:
That makes sense.  The reformat is quite a bit more extensive than I would have expected from a run spotless/edit/merge/edit/run spotless cycle.  Hopefully we don't get too many more situations like this one.